repos/system_upgrade/el7toel8/actors/checknonmountboots390/libraries/checknonmountboots390.py

```diff
@@ -0,0 +1,26 @@
+import os
+
+from leapp import reporting
+from leapp.libraries.common.config import architecture
+
+
+def perform_check():
+    if not architecture.matches_architecture(architecture.ARCH_S390X):
+        return
+
+    if os.path.ismount('/boot'):
+        return
+
+    data = [
+        reporting.Title('Found /boot is not on a separate partition'),
+        reporting.Summary((
+            'Due to some bug in the leapp code if /boot is not on a separate partition, the upgrade'
+            ' process erases the content of /boot. The problem could not be fixed in time and the upgrade'
+            ' has been inhibited to prevent data loss.'
```

Comment:
What about: "Due to a bug in the Leapp code, there is a situation when the upgrade process removes content of /boot when the directory is not on a separate partition and the system is running on S390x architecture. To avoid this from happening, we are inhibiting the upgrade process in this release until the issue has been fixed.